File: src/org/elixir_lang/structure_view/element/call_definition_by_name_arity/CallDefinitionbyNameArity.java
package org.elixir_lang.structure_view.element.call_definition_by_name_arity;

import com.intellij.openapi.util.Pair;
import org.elixir_lang.structure_view.element.CallDefinition;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

public interface CallDefinitionbyNameArity extends Map<Pair<String, Integer>, CallDefinition> {
    /**
     * Generates a {@link CallDefinition} for the given {@code nameArity} if it does not exist.
     *
     * The {@link CallDefinition} is
     * @param nameArity
     * @return pre-existing {@link CallDefinition} or new {@link CallDefinition} add to the {@code List<TreeElement>}
     */
    @NotNull
    CallDefinition putNew(@NotNull Pair<String, Integer> nameArity);
}

File: src/org/elixir_lang/structure_view/element/call_definition_by_name_arity/CallDefinitionByNameArity.kt
package org.elixir_lang.structure_view.element.call_definition_by_name_arity

import org.elixir_lang.NameArity
import org.elixir_lang.structure_view.element.call.Definition

interface CallDefinitionByNameArity : Map<NameArity, Definition> {
    /**
     * Generates a [Definition] for the given `nameArity` if it does not exist.
     *
     * The [Definition] is
     * @param nameArity
     * @return pre-existing [Definition] or new [Definition] add to the `List<TreeElement>`
     */
    fun putNew(nameArity: NameArity): Definition
}
